guineaPigFacts/src/main/java/com/factsdemo/guineaPigFacts/models/Fact.java

```diff
@@ -1,16 +1,30 @@
 package com.factsdemo.guineaPigFacts.models;
+
 import org.springframework.data.annotation.Id;
 import org.springframework.data.mongodb.core.mapping.Document;
 
-import java.util.Date;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Pattern;
+import javax.validation.constraints.Size;
+
+/**
+ * Class is mapped to the "fact" MongoDB collection
+ */
 
-//Class is mapped to the "fact" MongoDB collection
-@Document
+@Document(collection = "Fact")
 public class Fact {
     @Id
+    @Pattern(regexp = "^[0-9a-fA-F]{24}$")
     private String id;
+    @NotEmpty
+    @Size(min = 8)
+    @Pattern(regexp = ".*\\s.*")
     private String factValue;
-    private Date dateCreated;
+    @NotNull
+    @Pattern(regexp = "^[0-3][0-9]/[0-3][0-9]/(?:[0-9][0-9])?[0-9][0-9]$")
+    private String dateCreated;
```

Comment:
There was a bug that I could not figure out how to fix. When I created a new Date object for dateCreated in the Fact constructor the date was a null value. I changed it to a string to prevent that from happening again. 

I am trying to convert it back to Date as it is best practice. I'm currently going through StackOverflow trying to see if there's an issue using Date with MongoDB or if there's a special annotation I need to use with it.

Reply:
Makes sense.  If you have trouble figuring it out, let me know and we can take a look at it.

Reply:
Assigning a new Date instance outside of the constructor when the dateCreated variable is created seems to have fixed the problem: 
private Date dateCreated = new Date(); 
instead of Date dateCreated;
There must have been an order of operations error when creating the Date object and Fact object?

Reply:
That is really odd.  Doing the initialization in the constructor or on the field itself are both valid ways of initializing class instance members.  Maybe there was some other thing going on that subsequently set the dateCreated to null.    How are you creating instances of the Fact class?  Are you doing it manually or are you relying on Spring's serialization of the request body?

Reply:
I'm relying on Spring's deserialization of the request body. 

I thought that wouldn't be an issue since the date is not a parameter in the constructor. Therefore, when / if a user passes a null date it shouldn't matter since the date was being automatically set in the constructor.

Reply:
Been doing some reading and it looks like Spring uses the Jackson library for serialization/deserialization.

It looks like you have to annotate the constructor and constructor params with @JsonCreator and @JsonProperty respectively for Jackson to know which constructor to use for deserialization and which properties are associated with constructor params.

See this SO post for an example and explanation: https://stackoverflow.com/a/39123304

With deserialization libraries that I've used in the past (not Jackson though), the default constructor (the one without parameters) will always be used to deserialize an object unless you explicitly annotate it and tell the serialization library to use a non-default constructor.  If this is what was happening, then it would make sense that dateCreated was null because the constructor was never actually being invoked!